internal/pkg/manifest/validate.go

```diff
@@ -401,6 +401,15 @@ func (s ScheduledJobConfig) Validate() error {
 	return nil
 }
 
+// Validate returns nil if the pipeline manifest is configured correctly.
+func (p PipelineManifest) Validate() error {
```

Comment:
nit: @iamhopaul123 was there a reason why we don't want to have the receivers as pointers? since these are large structs, I wonder if we should make them pointers for performance reasons
```suggestion
func (p *PipelineManifest) Validate() error {
```